app/src/main/java/org/sagebionetworks/bridge/researchstack/CrfPrefs.java

```diff
@@ -54,6 +69,28 @@ public void setCustomSurveyCounter(int counter) {
         prefs.edit().putInt(KEY_CUSTOM_SURVEY_COUNTER, counter).apply();
     }
 
+    public boolean hasFirstSignInDate() {
+        return getFirstSignInDate() != null;
+    }
+
+    public DateTime getFirstSignInDate() {
+        String jsonString = prefs.getString(KEY_FIRST_SIGN_IN_DATE_TIME, null);
+        if (jsonString == null) {
+            return null;
+        }
+        return FORMATTER.parseDateTime(jsonString);
+    }
```

Comment:
Is it worth adding unit tests for get/setFirstSignInDate? At the very least, we should test that we're correctly parsing and preserving the timezone, since Joda is known to drop timezones while parsing.

Reply:
This code for get/setFirstSignInDate came directly from Gson's serializer/deserializer, so I figured I think it's safe to assume it's rock solid.